presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -3168,6 +3169,9 @@ public void testIn()
 
         String list =  Joiner.on(',').join(range(0, 20_000).boxed().iterator());
         assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (" + list + ")");
+
+        String listTimestamps = range(0, 2_000).asLongStream().mapToObj(i -> "TIMESTAMP '1970-01-01 00:01:0" + i / 1000 + "." + i % 1000 + "+00:00'").collect(Collectors.joining(", "));
+        assertQuery("SELECT x FROM (values TIMESTAMP '1970-01-01 01:01:00+01:00', TIMESTAMP '1970-01-01 01:01:00.99+01:00', TIMESTAMP '1970-01-01 00:01:00.999+00:00', TIMESTAMP '1970-01-01 00:01:00.999+02:00') t(x) WHERE x IN (" + listTimestamps + ")", "values TIMESTAMP '1970-01-01 01:01:00+01:00', TIMESTAMP '1970-01-01 01:01:00.99+01:00', TIMESTAMP '1970-01-01 00:01:00.999+00:00'");
```

Comment:
Put the test in `TestExpressionCompiler.testHugeIn` instead. At the top of `FunctionAssertions`, you can add `bound_timestamp_with_timezone`. That code there is old and doesn't use stream API, feel free to fix that.